From 98bdbb2cb31191b1afca440154ba2489eed70d3e Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Wed, 4 Jun 2014 15:38:38 +0800
Subject: [PATCH] sama5d3_xplained: set ENV_FAT_DEVICE_AND_PART intead of using
 number

We set ENV_FAT_DEVICE_AND_PART to "0". That specify the device number
and not specify the partition number. So when
  1. no partition table:
	take device 0 as a FAT file system.
  2. has parttion table:
	use device 0 partition 1.

For more details, please refer to the commit: 10a37fd7a4

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 include/configs/sama5d3_xplained.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/configs/sama5d3_xplained.h b/include/configs/sama5d3_xplained.h
index f72ab0b..861df2b 100644
--- a/include/configs/sama5d3_xplained.h
+++ b/include/configs/sama5d3_xplained.h
@@ -169,13 +169,15 @@
 				"bootz 0x22000000 - 0x21000000"
 #elif CONFIG_SYS_USE_MMC
 /* bootstrap + u-boot + env in sd card */
-#define CONFIG_ENV_IS_IN_MMC
-#define CONFIG_ENV_OFFSET	0x2000
-#define CONFIG_ENV_SIZE		0x1000
+#define CONFIG_ENV_IS_IN_FAT
+#define CONFIG_FAT_WRITE
+#define FAT_ENV_INTERFACE	"mmc"
+#define FAT_ENV_FILE		"uboot.env"
+#define FAT_ENV_DEVICE_AND_PART	"0"
+#define CONFIG_ENV_SIZE		0x4000
 #define CONFIG_BOOTCOMMAND	"fatload mmc 0:1 0x21000000 at91-sama5d3_xplained.dtb; " \
 				"fatload mmc 0:1 0x22000000 zImage; " \
 				"bootz 0x22000000 - 0x21000000"
-#define CONFIG_SYS_MMC_ENV_DEV	0
 #else
 #define CONFIG_ENV_IS_NOWHERE
 #endif
-- 
1.9.1

